core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt

```diff
@@ -436,8 +436,8 @@ internal class SyncStream(
         state: SyncStreamState,
     ): SyncStreamState {
         val batch = SyncDataBatch(listOf(data))
-        status.update { copy(downloading = true, downloadProgress = downloadProgress?.incrementDownloaded(batch)) }
         bucketStorage.saveSyncData(batch)
+        status.update { copy(downloading = true, downloadProgress = downloadProgress?.incrementDownloaded(batch)) }
```

Comment:
I think moving this down here may improve test reliability.

In particular with the two "sync progress after interrupted sync" tests, what could happen is that we update the state here, the test asserts that the state was updated and immediately closes the database (before we're able to save the batch). Then when we connect again, the test asserts that some data has already been downloaded but it's not the case due to us closing the database too early.

So let's hope that this improves the flaky tests we've seen 🤞